hs_tools_resource/utils.py

```diff
@@ -58,3 +62,48 @@ def get_SupportedSharingStatus_choices():
             ['Discoverable', 'Discoverable'],
             ['Private', 'Private'],
             ]
+
+
+def do_work_when_launching_app_as_needed(app_shortkey, res_shortkey, user):
+    """
+    check whether there are extra work needed to be done when the app is launched.
+    :param app_shortkey: shortkey of the app tool resource to be launched
+    :param res_shortkey: shortkey of the resource launching the app tool resource
+    :param user: requesting user
+    :return: error message if an exception is raised; otherwise, return None
+    """
+
+    # When app resource has iRODS federation target path and target resource defined as
+    # extended metadata, the resource needs to be pushed to the specified iRODS federation path
+    # in the specified iRODS storage resource.
+    try:
+        res = get_resource_by_shortkey(res_shortkey)
+        app_res = get_resource_by_shortkey(app_shortkey)
+
+        # check whether irods_path_key and irods_resc_key are added as extended metadata of the
+        # app tool resource, and if they are, pass them out as context variables for app launcher
+        # to push resource to federated iRODS path accordingly
+        irods_path_key = 'irods_federation_target_path'
+        irods_resc_key = 'irods_federation_target_resource'
+        filterd_app_obj = BaseResource.objects.filter(short_id=app_shortkey).filter(
+            extra_metadata__has_key=irods_path_key).filter(
+            extra_metadata__has_key=irods_resc_key).first()
+        if filterd_app_obj:
+            if not user.is_authenticated():
+                return "Only authorized users can launch the web app tool - Please sign in first."
+            irods_path = app_res.extra_metadata[irods_path_key]
+            irods_resc = app_res.extra_metadata[irods_resc_key]
+            istorage = res.get_irods_storage()
+            src_path = res.root_path
+            # delete all temporary resources copied to this user's space before pushing resource
```

Comment:
Should this be configurable in the metadata?  If files already exist, other apps might not want us deleting files and may appreciate the control.

Reply:
@sblack-usu This is a good consideration, however, this configuration would be app-specific, rather than a uniform flag to clean up temporary space or not. This specific configuration flag can go to web app extended metadata so each app can config this flag as desired. I think this can be added when such a use case arises, rather than adding it now when we don't have such a use case. In majority of cases, users would prefer HydroShare to clean up spaces for them, so let's go with this for now and add a configuration flag when such a use case arises.